File: go/base/models/square.go
package models

import "math"

type Square struct {
	side float64
}

func (Square) New(side float64) *Square {
	var dimOne = map[bool]float64{true: side, false: 1}[side > 0]
	return &Square{side: dimOne}
}

func (s Square) Area() float64 {
	return math.Pow(s.side, 2)
}

func (s Square) Perimeter() float64 {
	return 4 * s.side
}

File: cplusplus/src/base/models/square.cpp
#include <cmath>
#include <shapes/base/models.hpp>

namespace shapes::base::models {
Square::Square(double side) {
    const double dimOne = (side > 0) ? side : 1;
    this->side = dimOne;
}

auto Square::area() const -> double {
    return std::pow(side, 2);
}

auto Square::perimeter() const -> double {
    return 4 * side;
}
} // namespace shapes::base::models
